lib/util.js

```diff
@@ -403,8 +403,9 @@ function formatValue(ctx, value, recurseTimes) {
   // Some type of object without properties can be shortcutted.
   if (keys.length === 0) {
     if (typeof value === 'function') {
-      return ctx.stylize(`[Function${value.name ? `: ${value.name}` : ''}]`,
-                         'special');
+      const cName = value.constructor ? value.constructor.name : 'Function';
```

Comment:
There’s a `getConstructorOf` utility inside this file for dealing with some edge cases :)